WorkloadManagementSystem/Executor/JobScheduling.py

```diff
@@ -222,16 +222,12 @@ def optimizeJob( self, jid, jobState ):
     return self.__setJobSite( jobState, stageSites )
 
   def __applySiteFilter( self, sites, active = False, banned = False ):
-    filtered = list( sites )
-    if active:
-      for site in sites:
-        if site not in active:
-          filtered.remove( site )
-    if banned:
-      for site in banned:
-        if site in filtered:
-          filtered.remove( site )
-    return filtered
+    filtered = set( sites )
+    if active and isinstance( active, ( list, set, dict ) ):
+      filtered &= set( active )
+    if banned and isinstance( banned, ( list, set, dict ) ):
+      filtered -= set( banned )
+    return list( filtered )
 
```

Comment:
Sorry but I don't understand what is in your commit ;-)
I had understood that 'active' was the sites selected by the user, therefore requires an AND with where the files are. My PR is just a rewrite of what was there before, right?
I don't understand what this is supposed to do:

filtered = filtered - set( active ) | filtered